### ISSUE TITLE
Update config file to install Spark version 1.4.0

### ISSUE THREAD

**Issue by erond** [awslabs/emr-bootstrap-actions#123]
Can you please update the install config file to install Spark version 1.4.0? There are several bugfix also related to the deployment on EMR in this revision. For instance, I hope that https://issues.apache.org/jira/browse/SPARK-2669 will fix the problems I'm facing when trying to use the `HiveContext` within a Spark Streaming app deployed on EMR in YARN-cluster mode (Spark `v1.3.1e`, `-h` option, AMI 3.7.0), i.e. it can't connect to the Hive metastore. First of all the default classpath, although I'm using the `-h` option, doesn't include the `datanucleus-*` jars (missing looking at the Spark UI, environment section) and this causes CNF errors once the driver tries to use the `HiveContext`. Once manually added the classpath using `--driver-class-path /home/hadoop/spark/classpath/hive/*` as `spark-submit` optional argument, the needed jars seem to be available in the environment, but I get:

```
15/06/11 17:04:53 WARN conf.HiveConf: hive-default.xml not found on CLASSPATH
15/06/11 17:04:54 INFO metastore.HiveMetaStore: 0: Opening raw store with implemenation class:org.apache.hadoop.hive.metastore.ObjectStore
15/06/11 17:04:54 INFO metastore.ObjectStore: ObjectStore, initialize called
15/06/11 17:04:54 INFO DataNucleus.Persistence: Property datanucleus.cache.level2 unknown - will be ignored
15/06/11 17:04:54 INFO DataNucleus.Persistence: Property hive.metastore.integral.jdo.pushdown unknown - will be ignored
15/06/11 17:04:54 ERROR Datastore.Schema: Failed initialising database.
Unable to open a test connection to the given database. JDBC url = jdbc:mysql://localhost:3306/hive?createDatabaseIfNotExist=true, username = hive. Terminating connection pool (set lazyInit to true if you expect to start your database after your app). Original Exception: ------
java.sql.SQLException: Access denied for user 'hive'@'localhost' (using password: YES)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:1084)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:4232)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:4164)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:926)
        ...
```

**Note**: I also added `--driver-java-options -Dscala.usejavacp=true` and `--files /home/hadoop/hive/conf/hive-site.xml`

Thanks,
erond

**Comment by christopherbozeman** [comment]
I was able to get HiveContext working with metastore on masternode in yarn-cluster by doing:

* changed master hive-site.xml to reference internal fqdn of master
* granted hive user permission in mysql auth to allow for db connections from other hosts besides the localhost
* added the following entries to the spark-submit
--driver-class-path /home/hadoop/spark/lib/datanucleus-api-jdo-3.2.6.jar:/home/hadoop/spark/lib/datanucleu
s-core-3.2.10.jar:/home/hadoop/spark/lib/datanucleus-rdbms-3.2.9.jar:/home/hadoop/spark/classpath/emr/mysql-connector-java-5.1.30.jar:hive-site.xml --files /home/hadoop/spark/conf/
hive-site.xml

I have not tested this with Streaming, though.

**Comment by christopherbozeman** [comment]
The first build of Spark 1.4.0 is up as version 1.4.0.a (https://github.com/awslabs/emr-bootstrap-actions/blob/master/spark/VersionInformation.md).   You may request it by using the arguments '-v 1.4.0.a'.   Currently, it is not the default for the bootstrap action.

**Comment by gabrielcrowdtilt** [comment]
schweeet, demoing it this week.

**Comment by erond** [comment]
Great, thanks!

I was just able to successfully complete a Spark 1.4.0 batch job lasting 1 hour, 30 minutes with no issues.

**Comment by tayal8** [comment]
Hi i am new to Emr cluster with custom bootstrap. I want to setup emr cluster with spark 1.4. Can you please let me know how can i achieve this.

**Comment by jkleckner** [comment]
@tayal8, @christopherbozeman mentioned above already: ```You may request it by using the arguments -v 1.4.0.a. Currently, it is not the default for the bootstrap action.```

**Comment by tayal8** [comment]
Yes i did that only, the command which i ran was:

aws emr create-cluster --name "Test cluster" --ami-version 3.8 --use-default-roles --ec2-attributes KeyName=emr --applications  Name=Hive --instance-count 3 --instance-type m3.xlarge --bootstrap-actions Path=s3://support.elasticmapreduce/spark/install-spark,Args=["--v=1.4.0.a"]

But it terminates cluster giving me bootstrap failure.  Is there anything wrong in this command.

**Comment by christopherbozeman** [comment]
@tayal8 , the Args section would be `Args=[-v,1.4.0.a]`

**Comment by tayal8** [comment]
@christopherbozeman & @jkleckner : thanks it worked for me Args=[-v,1.4.0.a]. I am able to run spark 1.4.0.

**Comment by christopherbozeman** [comment]
sparkR is now included in the 1.4.0 build version 1.4.0.b
